web/packages/teleport/src/services/userEvent/types.ts

```diff
@@ -16,6 +16,11 @@
 
 // these constants are 1:1 with constants found in lib/usagereporter/web/userevent.go
 
+import * as prehogApi from 'gen-proto-js/prehog/v1alpha/teleport_pb';
+
+export type CtaEvents = prehogApi.CTA;
+export const CtaEvents = prehogApi.CTA;
```

Comment:
Exporting both type and value seems a bit weird (also I have never assigned `enum` to `const`😛), the more clean version would be:
```ts
export { CTA as CtaEvent } from 'gen-proto-js/prehog/v1alpha/teleport_pb';
```
Nit: it should be called `CtaEvent` (not `CtaEvents`), look at TS docs: https://www.typescriptlang.org/docs/handbook/enums.html#numeric-enums

Reply:
Oh, this is totally weird. Sorry, I should've added a comment there.

So, even though `export { CTA as CtaEvent } from ...` passes the type check and tests, it fails on build:

```
"CTA" is not exported by "../../../gen/proto/js/prehog/v1alpha/teleport_pb.js", imported by "../../../web/packages/teleport/src/services/userEvent/types.ts".
```

It looks like we can only import * from `gen-proto-js/prehog/v1alpha/teleport_pb`.

If we do something like this:

```typescript
import * as prehogApi from 'gen-proto-js/prehog/v1alpha/teleport_pb';

export const CtaEvents = prehogApi.CTA;
```

`CtaEvents` is a value, not a type, so we can't use it as a type in functions:

```typescript
captureCtaEvent(event: CtaEvents) { // err: 'CtaEvents' refers to a value, but is being used as a type here. Did you mean 'typeof CtaEvents'?ts(2749)
}
```

And if we do `event: typeof CtaEvents`, we can't use the enum values:

```typescript
userEventService.captureCtaEvent(CtaEvents.CTA_ACCESS_REQUESTS); // err: Argument of type 'import("...v1alpha/teleport_pb").CTA' is not assignable to parameter of type 'typeof CTA'.ts(2345)
```

So AFAIK the only way to do that is with that strange hack I found on [stack overflow](https://stackoverflow.com/a/73624285/11411938):

> This way, we get the best of both worlds - types and values. And TS will figure out which one it wants.

Reply:
Update: importing `gen-proto-js/prehog/v1alpha/teleport_pb` doesn't really work. It uses `require` and won't run in the browser.

I suppose there is a way to fix this (maybe generate the proto file with different flags)? But due to time constraints, I rather just copy the enum for now and go back to this later. Thoughts?

Reply:
Could you only leave a comment with TODO above the enum?